### ISSUE TITLE
request.CRITICAL Doctrine\DBAL\Exception\DriverException: FUNCTION strftime does not exist

### ISSUE THREAD

**Issue by groovenectar** [kimai/kimai#352]
Latest master version as of this report, fresh install.

Using MySQL for the database driver, and I think that may be part of the issue, `strftime` may be a SQLite thing.

When saving a new user, or editing a user, a 500 error happens and here is the log:

```
[2018-10-09 09:17:56] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\DriverException: "An exception occurred while executing 'SELECT SUM(k0_.rate) AS sclr_0, SUM(k0_.duration) AS sclr_1, strftime('%m', k0_.start_time) AS sclr_2, strftime('%Y', k0_.start_time) AS sclr_3 FROM kimai2_timesheet k0_ WHERE k0_.start_time > ? AND k0_.end_time IS NOT NULL AND k0_.user = ? GROUP BY sclr_3, sclr_2 ORDER BY sclr_3 DESC, sclr_2 ASC' with params [0, 2]:  SQLSTATE[42000]: Syntax error or access violation: 1305 FUNCTION kimai.strftime does not exist" at /home/kimai/kimai/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 126 {"exception":"[object] (Doctrine\\DBAL\\Exception\\DriverException(code: 0): An exception occurred while executing 'SELECT SUM(k0_.rate) AS sclr_0, SUM(k0_.duration) AS sclr_1, strftime('%m', k0_.start_time) AS sclr_2, strftime('%Y', k0_.start_time) AS sclr_3 FROM kimai2_timesheet k0_ WHERE k0_.start_time > ? AND k0_.end_time IS NOT NULL AND k0_.user = ? GROUP BY sclr_3, sclr_2 ORDER BY sclr_3 DESC, sclr_2 ASC' with params [0, 2]:\n\nSQLSTATE[42000]: Syntax error or access violation: 1305 FUNCTION kimai.strftime does not exist at /home/kimai/kimai/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:126, Doctrine\\DBAL\\Driver\\PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1305 FUNCTION kimai.strftime does not exist at /home/kimai/kimai/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:144, PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1305 FUNCTION kimai.strftime does not exist at /home/kimai/kimai/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:142)"} []
```

The new user will save, but it displays a 500 error.

I'm going to try redoing everything in SQLite, but it appears that something related to SQLlite may be hardcoded...

**Comment by groovenectar** [comment]
Confirmed that after switching to SQLite, this error does not happen

**Comment by kevinpapst** [comment]
Did you switch the database driver in between? You might have to execute `bin/console clear:cache` when changing the config from sqlite to mysql.
I am running the latest master with MySQL without problems.
If the user gets saved and you go back to the user overview and then try to edit its profile, does that work?

**Comment by groovenectar** [comment]
When I first set it up, I had put the MySQL configuration in .env before migrating the tables, etc., so I'm not sure if there may have been some initial SQLite references in there.

I just tried switching back, then clearing the cache.

I was able to create and edit a user with no error.

So, I'm not sure what happened! After the migration I ran this command during setup:

`bin/console cache:warmup --env=prod`

And it seems the clear cache at least once early on.

I guess I'll go ahead and close this... but may revisit the next time I do a fresh install with updating to MySQL config from the start...
